components/gitpod-protocol/src/util/timeout.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+/**
+ * A resettable timeout, based on an AbortController + setTimeout.
+ */
+export class Timeout {
```

Comment:
> clear function does not reset _timer

Thanks for the finding! fixed

> clear function does not abort existing _abortController seems unexpected?

This is expected, doing otherwise would defeat the purpose.

But I see your point, tests would be nice. I can try for 30mins tomorrow to come up with a) better comments and b) meaningful tests.
Sounds good?

Reply:
Added tests in https://github.com/gitpod-io/gitpod/pull/20175/commits/b92edee30e63265084fb8a327fffb0201527b727